xradar/io/backends/nexrad_level2.py

```diff
@@ -0,0 +1,1381 @@
+        """
+        Retrieve the unambiguous range of the requested scans.
+
+        Parameters
+        ----------
+        scans : list or None
+            Scans (0 based) for which the unambiguous range will be retrieved.
+            None (the default) will return the range for all scans in the
+            volume.
+
+        Returns
+        -------
+        unambiguous_range : ndarray
+            Unambiguous range (in meters) for the requested scans.
+
+        """
+        if scans is None:
+            scans = range(self.nscans)
+        # unambiguous range is stored in tenths of km, x100 for meters
+        return self._radial_sub_array(scans, "unambig_range") * 100.0
+
+    def get_data(self, moment, max_ngates, scans=None, raw_data=False):
+        """
+        Retrieve moment data for a given set of scans.
+
+        Masked points indicate that the data was not collected, below
+        threshold or is range folded.
+
+        Parameters
+        ----------
+        moment : 'REF', 'VEL', 'SW', 'ZDR', 'PHI', 'RHO', or 'CFP'
+            Moment for which to to retrieve data.
+        max_ngates : int
+            Maximum number of gates (bins) in any ray.
+            requested.
+        raw_data : bool
+            True to return the raw data, False to perform masking as well as
+            applying the appropiate scale and offset to the data.  When
+            raw_data is True values of 1 in the data likely indicate that
+            the gate was not present in the sweep, in some cases in will
+            indicate range folded data.
+        scans : list or None.
+            Scans to retrieve data from (0 based). None (the default) will
+            get the data for all scans in the volume.
+
+        Returns
+        -------
+        data : ndarray
+
+        """
+        if scans is None:
+            scans = range(self.nscans)
+
+        # determine the number of rays
+        msg_nums = self._msg_nums(scans)
+        nrays = len(msg_nums)
+        # extract the data
+        set_datatype = False
+        data = np.ones((nrays, max_ngates), ">B")
+        for i, msg_num in enumerate(msg_nums):
+            msg = self.radial_records[msg_num]
+            if moment not in msg.keys():
+                continue
+            if not set_datatype:
+                data = data.astype(">" + _bits_to_code(msg, moment))
+                set_datatype = True
+
+            ngates = min(msg[moment]["ngates"], max_ngates, len(msg[moment]["data"]))
+            data[i, :ngates] = msg[moment]["data"][:ngates]
+        # return raw data if requested
+        if raw_data:
+            return data
+
+        # mask, scan and offset, assume that the offset and scale
+        # are the same in all scans/gates
+        for scan in scans:  # find a scan which contains the moment
+            msg_num = self.scan_msgs[scan][0]
+            msg = self.radial_records[msg_num]
+            if moment in msg.keys():
```

Comment:
Would it make sense, to leave the CF encoding/decoding to xarray? We would need to add according attributes to the variables, though. I think we can take care of this in a follow up PR.